sky/clouds/kubernetes.py

```diff
@@ -202,7 +201,14 @@ def make_deploy_resources_variables(
         acc_type = k.accelerator_type if k.accelerator_type else None
 
         # Select image based on whether we are using GPUs or not.
-        image = self.IMAGE_GPU if acc_count > 0 else self.IMAGE_CPU
+        image_id = self.IMAGE_GPU if acc_count > 0 else self.IMAGE_CPU
+        # Get the container image ID from the service catalog.
+        # Note that currently we do not support custom images, so this condition
+        # will always be triggered. In the future we may want to get image_id
+        # from the resources object if it is set.
+        if image_id.startswith('skypilot:'):
+            image_id = service_catalog.get_image_id_from_tag(
+                image_id, clouds='kubernetes')
```

Comment:
```suggestion
        image_tag = self.IMAGE_GPU if acc_count > 0 else self.IMAGE_CPU
        # Get the container image ID from the service catalog.
        # Note that currently we do not support custom images, so this condition
        # will always be triggered. In the future we may want to get image_id
        # from the resources object if it is set.
        image_id = None
        if image_tag.startswith('skypilot:'):
            image_id = service_catalog.get_image_id_from_tag(
                image_tag, clouds='kubernetes')
```
Perhaps we should separate between `image_tag` and `image_id`?

Reply:
agreed, its a bit ambiguous here. However, since the image_id can also come from `resources` in the future, perhaps its best to leave `image_id` here?